src/backends/db2/CMakeLists.txt

```diff
@@ -1,14 +1,87 @@
-###############################################################################
-#
-# This file is part of CMake configuration for SOCI library
-#
-# Copyright (C) 2011 Denis Chapligin
-# Distributed under the Boost Software License, Version 1.0.
-# (See accompanying file LICENSE_1_0.txt or copy at
-# http://www.boost.org/LICENSE_1_0.txt)
-#
-###############################################################################
-soci_backend(DB2
-  DEPENDS DB2
-  DESCRIPTION "SOCI backend for DB2" AUTHORS "Denis Chapligin"
-  MAINTAINERS "Denis Chapligin")
+if (SOCI_DB2_AS_AVAILABLE)
+  set(REQUIRED_FLAG "")
+else()
+  set(REQUIRED_FLAG "REQUIRED")
+endif()
+
+find_package(DB2 ${REQUIRED_FLAG})
+
+if (NOT DB2_FOUND)
+  message(STATUS "Disabling DB2 backend as the required dependencies were not found")
+  set(SOCI_DB2 OFF CACHE BOOL "" FORCE)
+  return()
+endif()
+
+add_library(soci_db2
+  ${SOCI_LIB_TYPE}
+    "blob.cpp"
```

Comment:
Inconsistent indentation and why do we quote all these file names? This makes them less readable and doesn't bring anything, it's not like we're ever going to have spaces in the source file names (or at least only over my dead body).

Reply:
I consider it good practice to quote filenames as they can (in general) contain spaces and I dislike having only selected ones being quoted. But if you have a strong opinion on this, I can remove the quotes.